vaultclient/vaultclient.go

```diff
@@ -45,43 +45,72 @@ func New(accessKey, secretKey, sessionToken, endpoint, region string) *Vault {
 	}
 }
 
-// makeAWSRequest is a helper function that handles AWS v4 signing and HTTP requests
+// makeAWSRequest is a helper function that handles AWS v4 signing and HTTP requests with retry logic
 // This centralizes the common request logic used by all API operations
 func (c *Vault) makeAWSRequest(ctx context.Context, action string, formData url.Values) (*http.Response, error) {
 	formData.Set("Action", action)
 	formData.Set("Version", "2010-05-08")
 
 	requestBody := formData.Encode()
-	req, err := http.NewRequestWithContext(ctx, "POST", c.endpoint, strings.NewReader(requestBody))
-	if err != nil {
-		return nil, fmt.Errorf("failed to create request: %w", err)
-	}
 
-	req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
-	req.Header.Set("Host", req.URL.Host)
+	maxAttempts := 3 // Use retry similar to default config here : https://docs.aws.amazon.com/sdk-for-go/v2/developer-guide/configure-retries-timeouts.html
+	delay := 2 * time.Second
 
-	// Hash the request body payload for AWS signing
-	hash := sha256.Sum256([]byte(requestBody))
-	payloadHash := hex.EncodeToString(hash[:])
+	var lastErr error
+	for attempt := 0; attempt < maxAttempts; attempt++ {
+		fmt.Println("attempt", attempt)
+		req, err := http.NewRequestWithContext(ctx, "POST", c.endpoint, strings.NewReader(requestBody))
+		if err != nil {
+			return nil, fmt.Errorf("failed to create request: %w", err)
+		}
 
-	// Sign the request using AWS SDK signer
-	signer := v4.NewSigner()
-	creds, err := c.creds.Retrieve(ctx)
-	if err != nil {
-		return nil, fmt.Errorf("failed to retrieve credentials: %w", err)
-	}
+		req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+		req.Header.Set("Host", req.URL.Host)
 
-	err = signer.SignHTTP(ctx, creds, req, payloadHash, ServiceName, c.region, time.Now())
-	if err != nil {
-		return nil, fmt.Errorf("failed to sign request: %w", err)
-	}
+		// Hash the request body payload for AWS signing
+		hash := sha256.Sum256([]byte(requestBody))
+		payloadHash := hex.EncodeToString(hash[:])
 
-	resp, err := c.httpClient.Do(req)
-	if err != nil {
-		return nil, fmt.Errorf("failed to make request: %w", err)
+		// Sign the request using AWS SDK signer
+		signer := v4.NewSigner()
+		creds, err := c.creds.Retrieve(ctx)
+		if err != nil {
+			return nil, fmt.Errorf("failed to retrieve credentials: %w", err)
+		}
+
+		err = signer.SignHTTP(ctx, creds, req, payloadHash, ServiceName, c.region, time.Now())
+		if err != nil {
+			return nil, fmt.Errorf("failed to sign request: %w", err)
+		}
+
+		resp, err := c.httpClient.Do(req)
+		if err == nil {
+			fmt.Println("Response status:", resp.StatusCode)
```

Comment:
Maybe these logs could be removed, I'm not sure we want to display this in the tool, as it might not be parse-able or consistent with all APIs?